src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -192,7 +197,146 @@ private DenseMatrix initRhs(LfNetwork lfNetwork, EquationSystem equationSystem,
         return factorsByVarConfig;
     }
 
-    public List<SensitivityValue> analyse(Network network, List<SensitivityFactor> factors, OpenLoadFlowParameters lfParametersExt,
+    /**
+     * If the slack is distributed, we need to create a factor for every participating generator
+     *
+     * @param lfNetwork
+     * @param initialFactors
+     * @return
+     */
+    private List<SensitivityFactor> getAllFactors(LfNetwork lfNetwork, List<SensitivityFactor> initialFactors,
+                                                  LoadFlowParameters.BalanceType balanceType) {
+        List<SensitivityFactor> factors = new ArrayList<>();
+
+        Set<String> monitoredLines = new HashSet<>();
+        for (SensitivityFactor initialFactor : initialFactors) {
+            // if the factor if an injection then we will compute it with distribution, otherwise we just keep it as it is
+            if (initialFactor instanceof BranchFlowPerInjectionIncrease) {
+                BranchFlowPerInjectionIncrease branchFactor = (BranchFlowPerInjectionIncrease) initialFactor;
+                monitoredLines.add(branchFactor.getFunction().getBranchId());
+            } else {
+                factors.add(initialFactor);
+            }
+        }
+
+        switch (balanceType) {
+            case PROPORTIONAL_TO_GENERATION_P_MAX:
```

Comment:
@Djazouli Geoffory prefers that we do not use factors that whould remain in the API. So we have to fix that.